### PR TITLE
J1939 tp

### PR DESCRIPTION
**Pull request by battjt**
Would you consider if I have missed any tests or how I should better describe the tests?

### CODE REVIEW

File: src-test/org/etools/j1939_84/bus/j1939/J1939TPTest.java [hunk 1]
```diff
@@ -25,6 +26,18 @@
 
 public class J1939TPTest {
 
+    static private void assertPacketsEquals(Collection<Packet> expected, Collection<Packet> actual) {
+        Assert.assertEquals(expected.toString(), actual.toString());
+    }
+
+    static private void assertPacketsEquals(Packet expected, Packet actual) {
+        Assert.assertEquals(expected.toString(), actual.toString());
```
Comment (gumbelmj) [inline src-test/org/etools/j1939_84/bus/j1939/J1939TPTest.java]: it seems like it might be better to write equals() on packet, but this works.

Comment (battjt) [reply src-test/org/etools/j1939_84/bus/j1939/J1939TPTest.java]: equals() needs a context.  Does the timestamp count?  This is the exact comparison we need for these tests.

File: src/org/etools/j1939_84/bus/j1939/J1939TP.java [hunk 2]
```diff
@@ -371,7 +371,7 @@ public void sendDestinationSpecific(Packet packet) throws BusException {
             int packetsToSend = p2.get(1);
             if (packetsToSend == 0) {
                 if ((p2.get64() & 0x0000FFFFFFFFFFFFL) != 0x0000FFFFFFFFFFFFL) {
-                    warn("tp.CM_CTS \"hold he connection open\" should be: %04X", 0x0000FFFFFFFFFFFFL);
+                    warn("TP.CM_CTS \"hold he connection open\" should be: %04X", 0x0000FFFFFFFFFFFFL);
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/bus/j1939/J1939TP.java]: *hold THE connection open

**Comment by gumbelmj** [comment]
I think the test names are fine.  You might run the test with coverage. There are few lines and branches that aren't exercised.

File: src-test/org/etools/j1939_84/bus/j1939/J1939TPTest.java (hunk 1 above)
Comment (gumbelmj) [reply src-test/org/etools/j1939_84/bus/j1939/J1939TPTest.java]: Oh, right. The time of each packet is different. Nevermind

File: src-test/org/etools/j1939_84/bus/j1939/J1939TPTest.java [hunk 3]
```diff
@@ -49,27 +192,70 @@ public void testBam() throws BusException {
                 "18EBFF00 06 05 06 07 08 09 FF FF");
         try (EchoBus bus = new EchoBus(0xF9);
                 J1939TP tp = new J1939TP(bus, 0)) {
+            // bus.log(() -> String.format("BAM %6d:", (System.currentTimeMillis() -
```
Comment (gumbelmj) [inline src-test/org/etools/j1939_84/bus/j1939/J1939TPTest.java]: commented out code should be removed.

**Pull request merged**